### ISSUE TITLE
Improvements to local end-to-end parser testing

### ISSUE THREAD

**Issue by fojackson8** [globaldothealth/list#1633]
Parsers are end-to-end tested locally, but this testing process isn't perfect. Here are some of the improvements we'd ideally like:

1. Easier setup of local testing environment. Current issues include: 

a) Latest installation of aws-sam-cli (1.15.0) doesn't talk to docker container on local-host network, so need to use older version (0.53.0)
b) Local invocation of retrieval function is patchy/broken, requiring us to host test subsets of the data on s3.
c) Running parsing function of times out with connection errors such as:
`requests.exceptions.ReadTimeout: UnixHTTPConnectionPool(host='localhost', port=None): Read timed out. (read timeout=60)`

2. Spinning-up stack locally allows us to see data in the curator portal, but the map view is populated with stock data from mongodb, rather than actual data we upload at runtime. Ideally we want to use this local version of the curator portal to sense check cases we've just uploaded. This issue relates to  #1624 .

3. Default when setting up local mongodb server should be all cases deleted. Currently mongodb stores previous cases from previous test runs, making local spin-up take longer (up to 20 mins). Can get around this by deleting all cases before starting each time, but would be helpful to change default.

Part of the difficulty here is running a replica of the entire stack locally is hard, and when it breaks it needs fixes that aren't obvious to non-devs. So either we make this process seamless, improve FAQ/docs, or consider an alternative. One alternative could be for local testing of parsers to always be run on the same machine, which has everything setup correctly and we can just SSH in for local parser testing. However, this is problematic as doesn't scale beyond the team.

**Comment by fojackson8** [comment]
These are just some of the issues, feel free to add more and comments on these

**Comment by fojackson8** [comment]
Now found that e2e testing of parsers works whilst mongoimport of parsed cases breaks, which implies that the parsing_lib may have the wrong idea of the schema. 

This could be another issue to fix here. We need one uniform schema, and ideally would like clear error messages indicating how our parsed cases don't fit that schema, to make parser building and manual imports via mongoimport easier

**Comment by jim-sheldon** [comment]
Related to https://github.com/globaldothealth/list/issues/1932

**Comment by jim-sheldon** [comment]
Since this issue was authored a while ago and we've deployed a local parser testing environment, I will close this and we can use https://github.com/globaldothealth/list/issues/2199 for improvements.
